WordPress/gutenberg
# Widgets: Refactor the inspector in Widgets Customizer to use core's controls

## Description

Supersede #29755, close #30271.

I essentially rewrote the inspector by using core's controls section. Simplified the code a bit and moved most of the logics to `controls`. This also fixes the accessibility and UX issues of the previous attempt.
 
## How has this been tested?



1. Enable "Widgets blocks in Customizer" gutenberg experiment flag.
2. Go to Customize -> Widgets.
3. Inspect some blocks, it should open the inspector panel in a new section.
4. The focus should be moved to the back button of the inspector section.
5. Clicking on it or pressing <kbd>Escape</kbd> should close the inspector.
6. Closing the inspector should smoothly transition back to the editor and return focus back to the selected block.
7. It's possible to go to different widget areas and open their corresponding inspector panel.
8. Scrolling in the inspector should trigger the sticky header (See the screen recordings for more details.)
9. The inspector now has the correct title.

## Screenshots 

https://user-images.githubusercontent.com/7753001/113235395-890dbb80-92d5-11eb-91ae-106c8df0d164.mp4


## Types of changes




Bug fix

## Checklist:
